src/persist-client/src/read.rs

```diff
@@ -0,0 +1,224 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Read capabilities and handles
+
+use std::fmt::Debug;
+use std::marker::PhantomData;
+use std::num::NonZeroUsize;
+use std::time::Duration;
+
+use differential_dataflow::difference::Semigroup;
+use differential_dataflow::lattice::Lattice;
+use mz_persist_types::{Codec, Codec64};
+use serde::{Deserialize, Serialize};
+use timely::progress::{Antichain, Timestamp};
+use uuid::Uuid;
+
+use crate::error::{InvalidUsage, LocationError};
+
+/// An opaque identifier for a reader of a persist durable TVC (aka shard).
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct ReaderId(pub(crate) [u8; 16]);
+
+impl std::fmt::Display for ReaderId {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        std::fmt::Display::fmt(&Uuid::from_bytes(self.0), f)
+    }
+}
+
+/// A token representing one partition of a "snapshot" (the contents of a shard
```

Comment:
nit: sorry for being somewhat late with this, but: I don't know if "partition" is a good term for this. When I hear partition I think of things like hash-partitioning, which is related to sharding (and that term is also used already). Maybe _split_ is a good term for this thing? To me, it has fewer connotations of a "purposeful" partitioning of the data.

Feel free to ignore, though.

Reply:
split works for me!